js_modules/dagit/packages/ui/src/components/MiddleTruncate.tsx

```diff
@@ -14,51 +14,53 @@ import {calculateMiddleTruncation} from './calculateMiddleTruncation';
  *
  * When the DOM element resizes, the measurement and calculation steps will occur again.
  */
-export const MiddleTruncate: React.FC<{text: string}> = React.memo(({text}) => {
-  // Track the font style and target maximum width. `null` means no measurement has
-  // taken place yet.
-  const [truncatedText, setTruncatedText] = React.useState<string | null>(null);
-
-  // An element that renders the full text into the container, for the purpose of
-  // measuring the maximum available/necessary width for our truncated string.
-  const measure = React.useRef<HTMLDivElement>(null);
-
-  // Given the target font style and allotted width, calculate the largest possible middle-
-  // truncated string.
-  const calculateTargetStyle = React.useCallback(() => {
-    if (measure.current) {
-      setTruncatedText(calculateMiddleTruncatedText(measure.current, text));
-    }
-  }, [text]);
-
-  // Use a layout effect to trigger the process of calculating the truncated text, for the
-  // initial render.
-  React.useLayoutEffect(() => {
-    calculateTargetStyle();
-  }, [calculateTargetStyle]);
-
-  // If the container has just been resized, recalculate.
-  useResizeObserver(measure.current, () => {
-    calculateTargetStyle();
-  });
-
-  // Copy the full text, not just the truncated version shown in the DOM.
-  const handleCopy = React.useCallback(
-    (e: React.ClipboardEvent<HTMLDivElement>) => {
-      e.preventDefault();
-      const clipboardAPI = navigator.clipboard;
-      clipboardAPI.writeText(text);
-    },
-    [text],
-  );
-
-  return (
-    <Container onCopy={handleCopy} title={text}>
-      <span>{truncatedText}</span>
-      <MeasureWidth ref={measure}>{text}</MeasureWidth>
-    </Container>
-  );
-});
+export const MiddleTruncate: React.FC<{text: string; showTitle?: boolean}> = React.memo(
+  ({text, showTitle = true}) => {
```

Comment:
Added the `showTitle` prop to allow disabling the `title` attribute, e.g. in this case where we have a `data-tooltip` to handle the behavior.